### PR TITLE
Ehri 243

### PR DESCRIPTION
**Pull request by boychev**
- sort geonames locations by type so we can choose the best candidate by type when there are no related locations to disambiguate on
- add a new feature containing the phonetic encoding of the string for matching strings that sound similar with the gazetteer
- trainable phonetic encoding algorithm

### CODE REVIEW

File: xgapp-access-points/src/main/xgapp/application-resources/queries/geonames-meta.rq [hunk 1]
```diff
@@ -26,7 +24,9 @@ SELECT ?concept ?type ?latitude ?longitude ?ancestors ?population ?wikipediaLink
         ?concept geo:countryCode ?country.
         FILTER(?type = geo:A.ADM1 || ?type = geo:A.ADM1H
```
Comment (boyankukushev) [inline xgapp-access-points/src/main/xgapp/application-resources/queries/geonames-meta.rq]: `FILTER (?type in (geo:A.ADM1, geo:A.ADM1H, ...) || ?country in ("AL", "AT" ...)) .`
This variant seems faster for other queries.

File: xgapp-access-points/src/main/xgapp/application-resources/queries/geonames-labels.rq [hunk 2]
```diff
@@ -26,7 +24,9 @@ SELECT ?concept ?type ?literal WHERE {
         ?concept geo:countryCode ?country.
         FILTER(?type = geo:A.ADM1 || ?type = geo:A.ADM1H
```
Comment (boyankukushev) [inline xgapp-access-points/src/main/xgapp/application-resources/queries/geonames-labels.rq]: see the comment for `geonames-meta.rq`

File: xgapp-access-points/src/main/xgapp/application-resources/groovy/DoNothing.groovy [hunk 3]
```diff
@@ -0,0 +1,5 @@
+public static Set<String> workFlow(String funk, String input, String clazz) {
```
Comment (boyankukushev) [inline xgapp-access-points/src/main/xgapp/application-resources/groovy/DoNothing.groovy]: Is this one needed?

File: xgapp-access-points/pom.xml [hunk 4]
```diff
@@ -10,7 +10,6 @@
         <version>1.0-SNAPSHOT</version>
```
Comment (boyankukushev) [inline xgapp-access-points/pom.xml]: Use the `Major.Minor.Patch` versioning; 1.0.0-SNAPSHOT

File: typo-buster/pom.xml [hunk 5]
```diff
@@ -10,7 +10,6 @@
         <version>1.0-SNAPSHOT</version>
     </parent>
 
-    <groupId>com.ontotext.ehri</groupId>
     <artifactId>typo-buster</artifactId>
     <version>1.0-SNAPSHOT</version>
```
Comment (boyankukushev) [inline typo-buster/pom.xml]: 1.0.0-SNAPSHOT

File: access-point-atomizer/pom.xml [hunk 6]
```diff
@@ -10,7 +10,6 @@
         <version>1.0-SNAPSHOT</version>
     </parent>
 
-    <groupId>com.ontotext.ehri</groupId>
     <artifactId>access-point-atomizer</artifactId>
     <version>1.0-SNAPSHOT</version>
```
Comment (boyankukushev) [inline access-point-atomizer/pom.xml]: 1.0.0-SNAPSHOT

File: access-point-atomizer/pom.xml [hunk 7]
```diff
@@ -10,7 +10,6 @@
         <version>1.0-SNAPSHOT</version>
```
Comment (boyankukushev) [inline access-point-atomizer/pom.xml]: 1.0.0-SNAPSHOT

**Comment by stefan-enev** [comment]
Merging this for EHRI, if/when we start porting parts of this code to the Gazetteer we will have to fix a few bits.

**Pull request merged**